README.md

```diff
@@ -1,17 +1,215 @@
+If stopped is specified, then the site will be created and kept stopped.
+
+##### `name`
+
+The Name of the IIS site. Used for uniqueness. Will set the target to this value if target is unset.
+
+##### `physicalpath`
+
+The physical path to the IIS web site folder
+
+##### `applicationpool`
+
+The name of an ApplicationPool for this IIS Web Site
+
+##### `enabledprotocols`
+
+The protocols enabled for this site. If https is specified, http is implied. If no value is provided, then this setting is disabled
+
+##### `serviceautostart`
+
+Enables autostart on the specified website
+
+##### `serviceautostartprovidername`
+
+Specifies the provider used for service auto start. Used with :serviceautostartprovidertype.
+The <serviceAutoStartProviders> element specifies a collection of managed assemblies that Windows Process Activation Service (WAS) will load automatically when the startMode attribute of an application pool is set to AlwaysRunning. This collection allows developers to specify assemblies that perform initialization tasks before any HTTP requests are serviced.
+
+###### Example:
+
+```puppet
+iis_site { 'mysite'
+  ...
+  serviceautostartprovidername => "MyAutostartProvider",
+  serviceautostartprovidertype => "MyAutostartProvider, MyAutostartProvider, version=1.0.0.0, Culture=neutral, PublicKeyToken=426f62526f636b73"
+}
+```
+
+##### `serviceautostartprovidertype`
+
+Specifies the application type for the provider used for service auto start. Used with :serviceautostartprovider
+
+###### Example:
+
+```puppet
+iis_site { 'mysite'
+  ...
+  serviceautostartprovidername => "MyAutostartProvider",
+  serviceautostartprovidertype => "MyAutostartProvider, MyAutostartProvider, version=1.0.0.0, Culture=neutral, PublicKeyToken=426f62526f636b73"
+}
+```
+
+##### `preloadenabled`
+
+Enables loading website automatically without a client request first
+
+##### `defaultpage`
+
+Specifies the default page of the site
+
+##### `logformat`
+
+Specifies the format for the log file. When set to WSC, it can be used in conjunction with :logflags
+
+##### `logpath`
+
+Specifies the physical path to place the log file
+
+##### `logperiod`
+
+Specifies how often the log file should rollover
+
+##### `logtruncatesize`
+
+Specifies how large the log file should be before truncating it. The value must be in bytes. The value can be any size between '1048576 (1MB)' and '4294967295 (4GB)'.
+
+##### `loglocaltimerollover`
+
+Use the system's local time to determine for the log file name as well as when the log file is rolled over
+
+##### `logflags`
+
+Specifies what W3C fields are logged in the IIS log file. This is only valid when :logformat is set to W3C.
+
+## Limitations
+
+### Compatibility
+
+#### OS Compatibility
+This module is compatible only with `Windows Server 2008R2`, `Windows Server 2012` & `Windows Server 2012R2`. 
+
+#### Powershell Compatibility
+This module requires that `Powershell 3` or greater is installed on the system.
```

Comment:
Since this is not the default for Windows Server 2008/R2, it should be noted as part of the setup section as well.